jadx-gui/src/main/java/jadx/gui/utils/RelativePathTypeAdapter.java

```diff
@@ -0,0 +1,43 @@
+package jadx.gui.utils;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonToken;
+import com.google.gson.stream.JsonWriter;
+
+public class RelativePathTypeAdapter extends TypeAdapter<Path> {
+	private final Path basePath;
+
+	public RelativePathTypeAdapter(Path basePath) {
+		this.basePath = basePath;
```

Comment:
```suggestion
		this.basePath = Objects.requireNonNull(basePath);
```
Additional check to fast fail if null somehow leaked here